### PR TITLE
Allow flow adjustment based on temperature calibration

### PR DESCRIPTION
**Pull request by artursilva0**
Update yml and interface to add a new feature which allows temperature compensation on the flow rate for proportional valves with temperature sensors (back compatibility assured). Reserved registers were renamed and used for the new registers added, so I don't think that there is anything breaking, but please confirm.

### CODE REVIEW

**Review by MicBoucinha — APPROVED** [review]
LGTM!

File: Generators/Generators.csproj
Comment (bruno-f-cruz) [inline Generators/Generators.csproj:1]: Is this change necessary? If files dont need to be touched best practice is to just not fold them in the PR.

File: device.yml [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/harp-tech/reflex-generator/main/schema/device.json
 device: Olfactometer
 whoAmI: 1140
-firmwareVersion: "1.2"
-hardwareTargets: "1.0"
+firmwareVersion: "1.3"
+hardwareTargets: "1.1"
```
Comment (bruno-f-cruz) [inline device.yml:7]: Are all changes compatible with hardware 1.0 assuming that the firmware is updated to 1.3? If so, I would leave the hardwareTargets as 1.0 until we manage to solve https://github.com/harp-tech/reflex-generator/issues/42

File: device.yml [hunk 2]
```diff
@@ -325,18 +325,21 @@ registers:
     access: Write
     maskType: Channel3RangeConfig
     description: Selects the flow range for the channel 3.
-  Reserved0: &reserved
```
Comment (bruno-f-cruz) [inline device.yml:328]: I may be missing something, but all these changes are backwards compatible from the point of view of the API

**Review by bruno-f-cruz — CHANGES_REQUESTED** [review]

File: device.yml (hunk 1 above)
Comment (artursilva0) [reply device.yml:7]: Ok

File: device.yml (hunk 2 above)
Comment (artursilva0) [reply device.yml:328]: I think that all the changes are back compatible, so yes we can tag as 0.1.1.

File: Generators/Generators.csproj
Comment (artursilva0) [reply Generators/Generators.csproj:1]: Ok

**Review by bruno-f-cruz — COMMENTED** [review]
Looks good except for the extra line in the cproj.

**Comment by MicBoucinha** [comment]
The requested changes were updated in the latest commits. LGTM! @bruno-f-cruz do you agree?

File: Interface/Harp.Olfactometer/Harp.Olfactometer.csproj [hunk 3]
```diff
@@ -17,7 +17,7 @@
     <PackageLicenseFile>LICENSE</PackageLicenseFile>
     <PackageOutputPath>..\bin\$(Configuration)</PackageOutputPath>
     <TargetFrameworks>net462;netstandard2.0</TargetFrameworks>
-    <VersionPrefix>0.1.0</VersionPrefix>
+    <VersionPrefix>0.1.1</VersionPrefix>
```
Comment (glopesdev) [inline Interface/Harp.Olfactometer/Harp.Olfactometer.csproj:20]:
```suggestion
    <VersionPrefix>0.2.0</VersionPrefix>
```

Feels like this should be a bump to the minor version since new registers and features were added, no?

File: device.yml [hunk 4]
```diff
@@ -325,18 +325,21 @@ registers:
...
-    <<: *reserved
+    description: Temperature sensor reading value.
+  EnableTemperatureCalibration:
     address: 93
-  Reserved2:
-    <<: *reserved
+    type: U8
+    access: Read
+    description: Enable flow adjustment based on the temperature calibration.
+  UserTemperatureCalibration:
```
Comment (glopesdev) [inline device.yml:338]: What is the difference between `TemperatureValue` and `UserTemperatureCalibration`? Is this a reference value used to establish the adjustment automatically?

I am not very fond of adding the general term `User` here, since a "user" is very vaguely defined (and could be automated away in the future). It would be more helpful to describe here what exactly is expected this value to contain, e.g. if this is a reference value, we might consider changing this to `ReferenceTemperatureValue` and the description to more explicitly indicate what is the meaning of the values provided here.

File: device.yml [hunk 5]
```diff
@@ -325,18 +325,21 @@ registers:
...
     address: 94
+    type: U8
+    access: Read
+    description: User temperature calibration value.
```
Comment (glopesdev) [inline device.yml:342]: If appropriate, it would be helpful to have a more explicit description here.

File: Interface/Harp.Olfactometer/Harp.Olfactometer.csproj (hunk 3 above)
Comment (artursilva0) [reply Interface/Harp.Olfactometer/Harp.Olfactometer.csproj:20]: The rational was based on @bruno-f-cruz comment: " I think that prior to 1.x, we tend to use the patch from semver to indicate backward compatible. If these are indeed fully backwards compatible i would tag this release as 0.1.1."

File: device.yml (hunk 4 above)
Comment (artursilva0) [reply device.yml:338]: `TemperatureValue` is the actual value of the device temperature
`UserTemperatureCalibration` is used for calibration purposes only if the user decides to calibrate the device. This temperature calibration value is the value at which the calibration was done.

`User` is used to be consistent with former register description used for the calibration of the device, when performed by the user (the device always contains the factory calibration values, so `User` is used to distinguish this).

File: device.yml (hunk 5 above)
Comment (artursilva0) [reply device.yml:342]: Sure, will try to make it more clear.

File: device.yml (hunk 4 above)
Comment (artursilva0) [reply device.yml:338]: `TemperatureCalibrationValue` would be more clear?

Comment (glopesdev) [reply device.yml:338]: > TemperatureCalibrationValue would be more clear?

Definitely, that would be great if it works!

File: Interface/Harp.Olfactometer/Harp.Olfactometer.csproj (hunk 3 above)
Comment (glopesdev) [reply Interface/Harp.Olfactometer/Harp.Olfactometer.csproj:20]: Sounds great, in that case @artursilva0 changing this version to 0.2.0 and the name of the calibration register are the only two changes needed, everything else LGTM!

File: device.yml (hunk 4 above)
Comment (bruno-f-cruz) [reply device.yml:338]: It should be noted that adding the word User would make some sense given the current interface (https://github.com/harp-tech/device.olfactometer/blob/621022cf845f293f16a8558483cdf3ec015400f2/device.yml#L38).

Should we break this while we are still in 0.x?

Comment (artursilva0) [reply device.yml:338]: In fact, I've double checked and this register is shared for both the user and the factory calibration, so in the end `TemperatureCalibrationValue` seems more appropriated.  Do you agree @bruno-f-cruz , so that I can make the mods and merge these files?

**Comment by glopesdev** [comment]
@artursilva0 looks good, can you re-generate the interface scripts to make sure the name of the register class is updated on the C# side as well?

**Review by glopesdev — APPROVED** [review]

**Review by bruno-f-cruz — APPROVED** [review]
Sorry for the previous confusion on my part. Thanks for the PR!
